jni_gen_test/.gitignore

```diff
@@ -0,0 +1,16 @@
+# Files and directories created by pub.
```

Comment:
Can `jni_gen_test` be `jni_gen/test` instead?

It should be the unit tests and integration tests for `package:jni_gen`.

Reply:
jni_gen_test is a complete dart package with some tests after wrapper code is generated. Is it fine to nest a package like that in test/ folder?

Reply:
Why do you need a package?

* Any dependencies only use in tests can go in dev dependencies
* The wrapper code can be generated in the `setUp` of the test.

Is there something I'm missing?

Reply:
If you want full-package integration tests, you can take a look at how they are setup in:

https://github.com/dart-lang/ffigen/tree/master/example/c_json
https://github.com/dart-lang/ffigen/blob/master/test/large_integration_tests/large_test.dart#L62
https://github.com/dart-lang/ffigen/blob/master/test/large_integration_tests/_expected_cjson_bindings.dart

Reply:
> Why do you need a package?

The details are now slightly fuzzy in my head. But you can't generate code and use that in single test file. Test file that calls generated code should be loaded _after_setup is run.

Also code in `test/` is analyzed by `dart analyze` which may catch code depending on yet-to-be-generated sources.

I will see if there's another way to do this.

> large_test.dart

I think it only compares with expected bindings and does not run anything based on generated code. I wanted something end-to-end, that runs some test on generated code.

At this point jni_gen is at very early version and I think it's not viable to rely on exact comparison of generated bindings.

Reply:
> But you can't generate code and use that in single test file.

Correct, but you can (1) commit the generated code and generate code and use it in tests, and then (2) run the generator and compare the output string with the committed generated code in a second test.

In general, you cannot have test referring to generated files and not committing the generated file, even if you do it in a setup step.

> At this point jni_gen is at very early version and I think it's not viable to rely on exact comparison of generated bindings.

I think it's fine to update generated files every PR. This is what we have done with the `libclang` bindings in `ffigen` as well.

Reply:
Perhaps this is more of an example than a test? In that case you could put it in jni_gen/example? You can still run an example on github CI as if it was an integration test, it just won't be picked up by `dart test`